pkg/drivers/cdp/driver.go

```diff
@@ -135,6 +140,131 @@ func (drv *Driver) Open(ctx context.Context, params drivers.Params) (drivers.HTM
 	return LoadHTMLPage(ctx, conn, params)
 }
 
+func (drv *Driver) Parse(ctx context.Context, content []byte) (drivers.HTMLPage, error) {
```

Comment:
It feels that the code is copied from the ``Open`` function :)

Can we refactor the driver to avoid copying this?

Reply:
Most of the code is a boiled down version of LoadHTMLPage(...). But yes, I can move the code that is shared between Open(...) and Parse(...) to a new function and call that from both of them.